### PR TITLE
First version

### PR DESCRIPTION
**Pull request by raphaellondner-okta**
Shows how to use Okta /api/v1/users API through the C# SDK to access a user's profile properties and generate a CSV report of all the users in an Okta organization

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                }
+
+                fsOutput = new FileStream(strFilePath, FileMode.Create);
+                sw = new StreamWriter(fsOutput);
+
+                OktaClient oktaClient = new OktaClient(strOktaApiKey, new Uri(strOktaUrl));
+                UsersClient usersClient = oktaClient.GetUsersClient();
+                Uri nextPage = null;
+                PagedResults<User> users;
+                string headerLine = "Id,Login,Status,Created,Activated, Last Login Date, Last Updated Date, Password Changed Date, Status Changed, First Name,Last Name,Email,Secondary Email,Mobile Phone";
```
Comment (yiyangli-okta) [inline Program.cs]: nit: `headerLine` as `static` and `readonly`

File: Program.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    stopWatch.Start();
+                    foreach (var user in users.Results)
+                    {
+                        string line = string.Format("\"{0}\",\"{1}\",\"{2}\",\"{3}\",\"{4}\",\"{5}\",\"{6}\",\"{7}\",\"{8}\",\"{9}\",\"{10}\",\"{11}\",\"{12}\",\"{13}\"", user.Id, user.Profile.Login, user.Status, user.Created, user.Activated, user.LastLogin, user.LastUpdated, user.PasswordChanged, user.StatusChanged, user.Profile.FirstName, user.Profile.LastName, user.Profile.Email, user.Profile.SecondaryEmail, user.Profile.MobilePhone);
+                        List<string> unmappedProperties = user.Profile.GetUnmappedPropertyNames();
+                        StringBuilder sb = new StringBuilder();
+                        foreach (string unmappedProperty in unmappedProperties)
+                        {
+                            string sPropValue = user.Profile.GetProperty(unmappedProperty);
+                            sPropValue = sPropValue.Replace("\r\n", "");
```
Comment (yiyangli-okta) [inline Program.cs]: nit: I think you could `.Replace().Replace().Append()` and etc.

**Comment by yiyangli-okta** [comment]
I think it's good overall.

**Comment by yiyangli-okta** [comment]
I saw that .dll, like newton json is copied/imported as a static file. Is it okay we could use nuget packages here?

**Comment by raphaellondner-okta** [comment]
Thanks Yiyang! I've taken most of your feedback into account and will submit a new PR for you to review!

File: Program.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    TimeSpan ts = stopWatch.Elapsed;
+                    Console.WriteLine("Getting 100 users in {0} s", ts.TotalSeconds);
+
+                    stopWatch.Reset();
+                    stopWatch.Start();
+                    foreach (var user in users.Results)
+                    {
+                        string line = string.Format("\"{0}\",\"{1}\",\"{2}\",\"{3}\",\"{4}\",\"{5}\",\"{6}\",\"{7}\",\"{8}\",\"{9}\",\"{10}\",\"{11}\",\"{12}\",\"{13}\"", user.Id, user.Profile.Login, user.Status, user.Created, user.Activated, user.LastLogin, user.LastUpdated, user.PasswordChanged, user.StatusChanged, user.Profile.FirstName, user.Profile.LastName, user.Profile.Email, user.Profile.SecondaryEmail, user.Profile.MobilePhone);
+                        List<string> unmappedProperties = user.Profile.GetUnmappedPropertyNames();
+                        StringBuilder sb = new StringBuilder();
```
Comment (raphaellondner-okta) [inline Program.cs]: @yiyangli-okta Not sure how you would go about do this. unmappedProperties is only providing the names of the unmapped properties, not the actual values of profile properties. Or am I missing something here?

**Pull request closed without merging**